### DISCUSSION TITLE
RTL support

### DISCUSSION THREAD

**Discussion by themedleb** [outline/outline#1909] [Ideas]
Outline is awesome!! Never came across something like it, but it lacks RTL support, all it needs is the HTML markup to have `dir="auto"` because when not specified it defaults to `dir="left"` which is what's happening right now.

I hope my explanation is clear.

Edit: Oh I see someone already asked for this feature, sorry about that.
Also forgot to mention that when we specify the dir to auto (`dir="auto"`) both directions (left and right) work because the browser automatically detects the direction of content, so it's good for both directions at the same time and doesn't hurt any one of them, please have a look here: https://www.w3.org/International/questions/qa-html-dir#dirauto

Thank you in advance.

**Comment by tommoor** [comment]
This seems very do-able, if you're able I would appreciate you testing out the editor directly with the change you mentioned to see if RTL introduces any issues?

https://github.com/outline/rich-markdown-editor

**Reply by themedleb** [reply]
I tried to inject the dir attribute using the browser's inspector, and I used Arabic as an example.

So all I did was:
Added `dir="auto"` for the div that has the class: `class="ProseMirror"`, this is for the content's direction.
Same think can be done to the title (except of course the title will have a different HTML tag and class).

https://imgur.com/a/CKM2RvN

As you can see in the screenshots, everything works fine except for:
1. The Headings, the marks "H1" and "H2" and "H3" are on the foreground of the text, but I can write without a problem.
2. Some of the items needs a little padding from the right like the Checklist and the Quote and the placeholder that says: Type "/" to insert ...
3. In the table, the buttons I can click to add/remove rows and remove the table are displaced and not clickable anymore, but I can do everything else like writing and adding/removing columns ...
4. When I hover over the "Notice" messages, the list drop-down appears and covers the content of the notice.

So I think when these points are fixed, the chances for Outline's RTL support will go from 0% to 90%-95%.

And forgive me for this short answer, I couldn't find time to dive deep even though Outline worth all the time.

Hope this helps.

**Reply by tommoor** [reply]
This is great, it gives an idea of the scope of work required in the editor. If you're able at some point it would be really useful to create a "story" with realistic content that's RTL across the various types of nodes like you have done here and submit a PR: https://github.com/outline/rich-markdown-editor/blob/main/src/stories/index.stories.tsx

This would create a good testing ground for the fixes

**Reply by themedleb** [reply]
Oh, I'm so sorry, I have no idea about technologies like Node.js and React, except that they exist :sweat_smile:, I only have experience with HTML, CSS and a little bit of PHP, that's why I tried to help with only what I can.
